exporter/datadogexporter/utils/trace_helpers.go

```diff
@@ -132,3 +133,47 @@ func NormalizeServiceName(service string) string {
 
 	return s
 }
+
+// From: https://github.com/DataDog/datadog-agent/blob/a6872e436681ea2136cf8a67465e99fdb4450519/pkg/trace/traceutil/trace.go#L27
+// GetRoot extracts the root span from a trace
+func GetRoot(t *pb.APITrace) *pb.Span {
+	// That should be caught beforehand
+	spans := t.GetSpans()
+	if len(spans) == 0 {
+		return nil
+	}
+	// General case: go over all spans and check for one which matching parent
+	parentIDToChild := map[uint64]*pb.Span{}
+
+	for i := range spans {
+		// Common case optimization: check for span with ParentID == 0, starting from the end,
+		// since some clients report the root last
+		j := len(spans) - 1 - i
+		if spans[j].ParentID == 0 {
+			return spans[j]
+		}
+		parentIDToChild[spans[j].ParentID] = spans[j]
+	}
+
+	for i := range spans {
+		_, ok := parentIDToChild[spans[i].SpanID]
+
+		if ok {
+			delete(parentIDToChild, spans[i].SpanID)
+		}
+	}
+
+	// Here, if the trace is valid, we should have len(parentIDToChild) == 1
+	// if len(parentIDToChild) != 1 {
+	//	// log.Debugf("Didn't reliably find the root span for traceID:%v", t[0].TraceID)
+	// }
+
```

Comment:
Either remove
```suggestion
```
or pass the zap logger here

Reply:
turns out we don't really have _any_ good debug logging in trace translation code, but should be something we emit a debug log for. i removed the code, but I'd like to keep a small comment here for future work, since we should add in some debug logs but its probably a separate pr since it requires modifying a number of function signatures